### DISCUSSION TITLE
UI not refreshing automatically

### DISCUSSION THREAD

**Discussion by kowshik1234** [zauberzeug/nicegui#1896] [Q&A]
### Question

My application is simple, 

1. Read a text file
2. Parse the last line of the txt file which will be updated periodically every few seconds and extract the number.
3. Display the number on the UI without disturbing previously displayed numbers. 

I can't use any buttons or any other UI for a manual refresh, the update on the UI should happen as soon as the TXT file is updated with a new line. 

Can someone guide me through this please? I tried ui.refreshable and ui.refresh() options but nothing worked as per my required.

Thanks

**Comment by rodja** [comment]
I would use `ui.timer` to regularly check the file. Make sure you use aiofiles or similar to [not block the UI](https://github.com/zauberzeug/nicegui/wiki/FAQs#why-is-my-long-running-function-blocking-ui-updates) while reading the file. The result can then simply be pushed to a `ui.log`, added to a `ui.scroll_area` or similar.

**Reply by falkoschindler** [reply]
Note that you only need to check the last-modified time of the file, not the whole file content.

**Comment by falkoschindler** [comment]
Or you use something like https://github.com/samuelcolvin/watchfiles to watch for file changes.
